### PR TITLE
fix: Regression on null PK inserts on views for headers only representation

### LINKED ISSUE #1832: Regression: Nightly throws 500 "Unexpected NULL"
### Environment

* PostgreSQL version: 13.2
* PostgREST version: nightly (d3a8b5f6e1f541c5c6cde9398a7ba6a95cdbdb88)
* Operating system: FreeBSD 13

### Description of issue

There's a regression after 343e41c51d6cad33383043abb4ef15dc54a32245

PostgREST throws 500 "Unexpected NULL" instead 201 "Created".

@wolfgangwalther, why had you removed null PK value processing on this? https://github.com/PostgREST/postgrest/blob/6670a3214bc2f74eac10ff32eb69a38db3807eaf/src/PostgREST/Query/SqlFragment.hs#L174

### Testcase
```PLpgSQL
DROP VIEW IF EXISTS test_bug;
DROP TABLE IF EXISTS test_a;
DROP TABLE IF EXISTS test_b;

CREATE TABLE test_b
(
    id int,

    PRIMARY KEY (id)
);

CREATE TABLE test_a
(
    id   int,
    b_id int,

    PRIMARY KEY (id),
    FOREIGN KEY (b_id) REFERENCES test_b
);

CREATE OR REPLACE VIEW test_bug AS
SELECT a.id, b.id AS b_id
FROM test_a AS a
     LEFT JOIN test_b AS b ON a.b_id = b.id;

CREATE RULE test_bug_insert AS ON INSERT TO test_bug DO INSTEAD
    INSERT INTO test_a(id, b_id) VALUES (new.id, new.b_id) RETURNING id, b_id;
```

```HTTP
POST http://postgrest/test_bug
Content-Type: application/json

{"id": 1}

HTTP/1.1 500 Internal Server Error
...

{
  "message": "Row error: Wrong value parser used",
  "details": "Unexpected NULL"
}
```

### PR DESCRIPTION
**Pull request by laurenceisla**
Closes #1832

Added the PR #1855 missing tests.

### CODE REVIEW

File: test/Feature/InsertSpec.hs [hunk 1]
```diff
@@ -537,3 +537,13 @@ spec actualPgVersion = do
             , matchHeaders = [ "Location" <:> "/compound_pk_view?k1=eq.1&k2=eq.test"
                              , "Content-Range" <:> "*/*" ]
             }
+
+      it "should not throw and return location header when a PK is null" $
```
Comment (steve-chavez) [inline test/Feature/InsertSpec.hs:541]: I'm really puzzled as to how a PK can be null. Perhaps we can add a short comment of how that happens?

Comment (laurenceisla) [reply test/Feature/InsertSpec.hs:541]: For instance, in the use case exemplified in this test, PostgREST is recognizing both the `id` and `sponsor_id` columns of the VIEW as PRIMARY KEYS. In this case, a `competitor` without a `sponsor` is possible, this means that the column `sponsor_id` (identified as a PK in the VIEW) could have a NULL value.

With this regression fix, at the moment the Location Header is built, PostgREST handles this NULL value for the `sponsor_id` and returns the header correctly.

Comment (steve-chavez) [reply test/Feature/InsertSpec.hs:541]: Cool. So it's the inferred PKs on the VIEW that are causing the issue - 2 PKs for this view, and one of them is null.

File: test/Feature/InsertSpec.hs [hunk 2]
```diff
@@ -537,3 +537,13 @@ spec actualPgVersion = do
...
+        request methodPost "/test_null_pk_competitors_sponsors" [("Prefer", "return=headers-only")]
+            [json|{"id":1}|]
+          `shouldRespondWith`
+            ""
+            { matchStatus  = 201
+            , matchHeaders = [ "Location" <:> "/test_null_pk_competitors_sponsors?id=eq.1&sponsor_id=is.null"
```
Comment (steve-chavez) [inline test/Feature/InsertSpec.hs:547]: Just to clarify, `test_null_pk_competitors_sponsors` doesn't have`sponsor_id` as a pk column:

```sql
CREATE TABLE competitors (
    id integer primary key,
    sponsor_id integer references sponsors(id),
    full_name varchar(40) not null
);

CREATE VIEW test_null_pk_competitors_sponsors  AS
SELECT c.id, s.id as sponsor_id
FROM competitors c
    LEFT JOIN sponsors s on c.sponsor_id = s.id;
```

So the above would be a bug.

**Pull request merged**